Doc/c-api/typehints.rst

```diff
@@ -0,0 +1,41 @@
+.. highlight:: c
+
+.. _typehintobjects:
+
+Objects for Type Hinting
+------------------------
+
+Various built-in types for type hinting are provided.  Currently,
+two types exist -- :ref:`GenericAlias <types-genericalias>` and
+:ref:`Union <types-union>`.  Only ``GenericAlias`` objects are currently exposed
+when including :file:`Python.h`.
+
+.. c:function:: PyObject* Py_GenericAlias(PyObject *origin, PyObject *args)
+
+   Return a :ref:`GenericAlias <types-genericalias>` object on success,
+   ``NULL`` on failure.  Equivalent to calling the Python class
+   :class:`types.GenericAlias`.  *origin* and *args* arguments set the
+   ``GenericAlias``\ 's ``__origin__`` and ``__args__`` attributes respectively.
+   *origin* should be a type, and *args* **must** be a :c:type:`PyTupleObject`.
```

Comment:
I don't think that's true; the `setup_ga` method automatically packs anything that's not a tuple into a 1-tuple. Python itself passes a non-tuple here: https://github.com/python/cpython/blob/bb3e0c240bc60fe08d332ff5955d54197f79751c/Objects/abstract.c#L182

Reply:
Thanks for the catch and pointing this out. I should read the code more carefully next time.